### PR TITLE
[Mobile app] Enhancements to Insights

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Adds favourite place button, air quality number and share to insights page.
- [x] Adds widget  testing to insights page.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

#### Screenshots (optional)

Screenshot_1683212592

### CODE REVIEW

File: mobile/lib/widgets/dialogs.dart [hunk 1]
```diff
@@ -316,6 +316,109 @@ void showSnackBar(
   ScaffoldMessenger.of(context).showSnackBar(snackBar);
 }
 
+void showFavouritePlaceSnackBar(
+  BuildContext context,
+  AirQualityReading airQualityReading, {
+  int durationInSeconds = 2,
+}) {
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/dialogs.dart:323]: Favourite places snack bar

File: mobile/test/screens/insights/insights_widgets_test.dart [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+      dateTime: DateTime.now(),
+      pm2_5: 5,
+      healthTips: List.generate(
+        5,
+        (index) => healthTip,
+      ),
+      airQuality: Pollutant.pm2_5.airQuality(5),
+    );
+    name = "Makerere";
+  });
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply mobile/test/screens/insights/insights_widgets_test.dart:35]: Thanks @NoahNsimbe for these tests, much much welcome! :)

**Review by Baalmart — APPROVED** [review]
Thank you @NoahNsimbe ! This LGTM

I also noticed one item in the description is not checked, just an oversight or work in progress?

**Comment by NoahNsimbe** [comment]
> Thank you @NoahNsimbe ! This LGTM
> 
> I also noticed one item in the description is not checked, just an oversight or work in progress?

Thanks @Baalmart. I added the widget tests.

File: mobile/test/screens/insights/insights_widgets_test.dart [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+  group('Insight AirQuality Widget tests', () {
+    testWidgets('Tests none empty insight', (tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: InsightAirQualityWidget(insight, name: name),
+        ),
+      );
```
Comment (NoahNsimbe) [inline mobile/test/screens/insights/insights_widgets_test.dart:43]: Widget tests are here @Baalmart

**Review by BenjaminSsempala — APPROVED** [review]
LGTM

**Review by Mnoble-19 — APPROVED** [review]

**Pull request merged**